### PR TITLE
[Excel] (Custom Functions) Update Custom Functions 1.3 error handling documentation

### PR DESCRIPTION
**Pull request by alison-mk**
[Internal work item](https://office.visualstudio.com/OC/_workitems/edit/5008206).

Note to reviewers: The [Output an array with an error](https://review.docs.microsoft.com/office/dev/add-ins/excel/custom-functions-errors?branch=pr-en-us-2847#output-an-array-with-an-error) code sample works in Script Lab. Try it out!

### CODE REVIEW

**Comment by alison-mk** [comment]
@mscharlock I'm uncertain how to format the output in this sentence: 
> For example, a custom function could output the array `[1],[#NUM!],[3]`.
Please take a look and let me know what you think. Line 62 in custom-function-errors.md. 
Thanks!

**Review by grangeryy — APPROVED** [review]

File: docs/excel/custom-functions-json.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
-ms.date: 12/22/2020
-description: 'Define JSON metadata for custom functions in Excel and associate your function ID and name properties.'
 title: Manually create JSON metadata for custom functions in Excel
+description: 'Define JSON metadata for custom functions in Excel and associate your function ID and name properties.'
```
Comment (davidchesnut) [inline docs/excel/custom-functions-json.md:3]: don't forget to update the ms.date

File: docs/excel/custom-functions-errors.md [hunk 2]
```diff
@@ -58,12 +57,50 @@ let error = new CustomFunctions.Error(CustomFunctions.ErrorCode.invalidValue, "T
...
+  // Return the results of the first and third parameter calculations and a #NUM! error in place of the second result. 
+  return [[firstResult], [secondResult], [thirdResult]];
+}
+```
+
+### Errors as custom function inputs
+
+A custom function can evaluate even if the input range contains an error. For example, a custom function can take the range **A2:A7** as an input, even if **A6:A7** contains an error.
+
+To process inputs that contain errors, a custom function must have the JSON metadata property `allowErrorForDataTypeAny` set to `true`. See [Manually create JSON metadata for custom functions](custom-functions-json.md#metadata-reference) for more information.
```
Comment (davidchesnut) [inline docs/excel/custom-functions-errors.md:94]: Will these be autogenerated for you, or do you always have to manually set this in the JSON metadata?

File: docs/excel/custom-functions-errors.md [hunk 3]
```diff
@@ -58,12 +57,50 @@ let error = new CustomFunctions.Error(CustomFunctions.ErrorCode.invalidValue, "T
 throw error;
 ```
 
-## Use try-catch blocks
+### Output an array with an error
```
Comment (davidchesnut) [inline docs/excel/custom-functions-errors.md:60]: Does this apply when your custom function returns dynamic arrays? If so might be more clear to title this as "Handle an error when working with dynamic arrays" or similar.

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Cool feature. Please take a look at my review comments. Thanks!

File: docs/excel/custom-functions-errors.md (hunk 2 above)
Comment (alison-mk) [reply docs/excel/custom-functions-errors.md:94]: This question is answered in the !IMPORTANT note on line 96. Does that note lack clarity? Do you think that note should be placed in a different location?

File: docs/excel/custom-functions-json.md (hunk 1 above)
Comment (alison-mk) [reply docs/excel/custom-functions-json.md:3]: I did update the ms.date, see line 4.

**Review by davidchesnut — APPROVED** [review]
Looks great!

File: docs/excel/custom-functions-errors.md (hunk 2 above)
Comment (davidchesnut) [reply docs/excel/custom-functions-errors.md:94]: I didn't notice the note, my bad. The note is good where it is. thanks!

**Review by lindalu-MSFT — APPROVED** [review]
Looks good! No changes requested!

**Pull request merged**